lib/services/github.ts

```diff
@@ -488,42 +504,50 @@ export class GithubService extends WorkerClient<string> implements ServiceListen
 	 * @returns  Promise fulfilled once authentication has occurred.
 	 */
 	protected authenticate(): Promise<void> {
-		// Initialise JWTs
-		const privatePem = new Buffer(this.pem, 'base64').toString();
-		const payload = {
-			exp: Math.floor((Date.now() / 1000)) + (10 * 50),
-			iat: Math.floor((Date.now() / 1000)),
-			iss: this.integrationId
-		};
-		const jwToken = jwt.sign(payload, privatePem, { algorithm: 'RS256' });
-		const installationsOpts = {
-			headers: {
-				Accept: 'application/vnd.github.machine-man-preview+json',
-				Authorization: `Bearer ${jwToken}`,
-				'User-Agent': 'request'
-			},
-			json: true,
-			url: 'https://api.github.com/integration/installations'
-		};
-
-		return request.get(installationsOpts).then((installations) => {
-			// Get the URL for the token.
-			const tokenUrl = installations[0].access_tokens_url;
+		// Authentication depends on App or PAT.
+		let tokenPromise: Promise<AuthToken>;
 
-			// Request new token.
-			const tokenOpts: any = {
+		if (this.appId) {
+		// Initialise JWTs
+			const privatePem = new Buffer(this.pem, 'base64').toString();
+			const payload = {
+				exp: Math.floor((Date.now() / 1000)) + (10 * 50),
+				iat: Math.floor((Date.now() / 1000)),
+				iss: this.appId
+			};
+			const jwToken = jwt.sign(payload, privatePem, { algorithm: 'RS256' });
+			const appOpts = {
 				headers: {
 					Accept: 'application/vnd.github.machine-man-preview+json',
 					Authorization: `Bearer ${jwToken}`,
 					'User-Agent': 'request'
 				},
 				json: true,
-				method: 'POST',
-				url: tokenUrl
+				url: 'https://api.github.com/app/installations'
 			};
 
-			return request.post(tokenOpts);
-		}).then((tokenDetails) => {
+			tokenPromise = request.get(appOpts).then((apps) => {
+				// Get the URL for the token.
+				const tokenUrl = apps[0].access_tokens_url;
```

Comment:
Nope, you'll always get an empty array. However, the `apps[0]` will throw if it's empty, and we rely on that to stop the ProcBot run, as it shows that there's something wrong with the App details (essentially it's an invalid VersionBot instance).